assets/js/blocks/featured-product/block.js

```diff
@@ -161,7 +161,7 @@ class FeaturedProduct extends Component {
 
 		return (
 			<Placeholder
-				icon={ <IconStar /> }
+				icon={ <Icon icon="star-filled" /> }
```

Comment:
This can just be the string "star-filled", since it's now a dashicon it'll do the `Dashicon` wrapping by default. Same with the other Placeholder below, and the `registerBlockType` in the other file.